### DISCUSSION TITLE
Running asciinema in headless macOS CI (e.g. GitHub Actions)?

### DISCUSSION THREAD

**Discussion by hartwork** [asciinema/discussions#247] [General]
Hi!

I have run into [Create headless recording with `setsid`](https://asciinema.org/a/615308) just today. Command `setsid` seems to be available on macOS through…
```bash
brew install util-linux
export PATH="${PATH}:/usr/local/opt/util-linux/bin"
```
…but running `setsid asciinema ……` fails with error…

> Error opening terminal: unknown.

…on macOS in practice, so the `setsid` approach seems to be limited to Linux?

I've been using [this `headless.py` wrapper of mine](https://gist.github.com/hartwork/ef18f3d80dad9f13bbecb8d0f8456b87) in macOS CI so far and it works, but I wonder about simpler or more streamlined solutions that you may already be aware of. Any ideas?

Thanks in advance and best, Sebastian

**Comment by ku1ik** [comment]
Yeah, I suppose we could implement `--headless` option for `asciinema rec` command to force it to run in headless mode.

**Comment by ku1ik** [comment]
asciinema CLI 3.0 will include the `--headless` option. If you'd like to test this then you can get the 3.0 RC2 from here: https://github.com/asciinema/asciinema/releases/tag/v3.0.0-rc.2
